backend/src/db/cache_storage.py

```diff
@@ -34,28 +38,187 @@ def __init__(self):
             )
         except Exception as e:
             logger.error("Could not connect to redis server: %s" % e)
-            raise Exception(f"Could not connect to redis server: {e}")
+            raise ValueError(f"Could not connect to redis server: {e}")
 
-    def set_value(self, key, value, expiration: Optional[int] = None):
+        self.pubsub = self.cache.pubsub()
+
+    def set_value(self, key: Any, value: Any, expiration: Optional[int] = None) -> None:
+        """
+        Set a key-value pair in the cache with an optional expiration time.
+
+        Args:
+            key (Any): The key to set in the cache.
+            value (Any): The value to set in the cache.
+            expiration (Optional[int]): The expiration time in seconds (default is None).
+
+        Returns:
+            None
+        """
         if expiration:
             self.cache.set(key, value, expiration)
         else:
             self.cache.set(key, value, self.expiration_time)
 
-    def get_value(self, key):
+    def get_value(self, key: Any) -> Optional[Awaitable]:
+        """
+        Get the value associated with the given key from the cache.
+
+        Args:
+            key (Any): The key to retrieve the value for.
+
+        Returns:
+            Optional[Awaitable]: The value associated with the key, or None if key is not found.
+        """
         value = self.cache.get(key)
         if value is not None:
             return value
         return None
 
-    def delete_value(self, key):
+    def delete_value(self, key: Any) -> None:
+        """
+        Delete the key-value pair from the cache.
+
+        Args:
+            key (Any): The key to delete from the cache.
+
+        Returns:
+            None
+        """
         self.cache.delete(key)
 
-    def publish(self, channel, message):
-        pass
+    def publish_signal(
+        self,
+        pattern: bytes | str | memoryview,
+        serialized_data: bytes | memoryview | str | int | float,
+    ) -> Awaitable:
```

Comment:
This is wrong, Awaitable means that action is asynchronous. It means that if you want to get value from this method you should do `sth = await publish_signal()` which is not True.